x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_section.test.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { AnalyticsSection } from './';
+
+describe('AnalyticsSection', () => {
+  it('renders', () => {
+    const wrapper = shallow(
+      <AnalyticsSection title="Lorem ipsum" subtitle="Dolor sit amet.">
+        <div data-test-subj="HelloWorld">Test</div>
+      </AnalyticsSection>
+    );
+
+    expect(wrapper.find('h2').text()).toEqual('Lorem ipsum');
```

Comment:
I'm a big fan of using `const`s for pass-through values

Reply:
I'm not. I've mentioned this before, and will continue to do so, but I **very** strongly prefer using static output and not variables in our test assertions.

1. Performance is a non-issue in tests. Developer understanding is paramount and (in my opinion) is the foremost goal when it comes to writing unit tests. Unit tests should serve as living documentation as to the what, how, why, and edge cases of your code; another developer should be able to read _only_ your test file and fully understand what the source code does.

2. With that understanding in mind, tests should be written for human readability and not for either compilation time, abstraction, or ease of updating. If your source code changes, your tests should typically change as well, as should documentation.

Consider the developer experience of the following:

```js
const expectedValue = 'supercalifragilisticexpialidocious';

// …

// … a bunch of lines between your abstracted var and the test

// …

it('has an expected value', () => {
  expect(output).toEqual(expectedValue);
});
```

You're asking a developer to either hold in their memory the expected value between its declaration and it being used or you're asking them to find/scroll up and down your file. My question is: **why**? Why have this extra intermediate step or friction instead of assertions that read clearly and explicitly as-is, i.e.:
```js
expect(output).toEqual('supercalifragilisticexpialidocious');
``` 

Now, consider the following worse example (it's contrived, but hopefully illustrative):
```js
describe('someFn', () => {
  it('capitalizes strings and appends the word YOLO', () => {
    const input = 'supercalifragilisticexpialidocious';
    expect(someFn(input)).toEqual(input.toUpperCase() + 'YOLO');
  });
});
```
I consider this a particularly bad antipattern because the developer is **literally just rewriting the logic of the function**. That is NOT the point of a unit test, it doesn't scale the more complex the logic gets, and worse, it does not aid developer understanding (you might as well just go read the source code instead of reading the test). Consider this static example:
```js
describe('someFn', () => {
  it('capitalizes strings and appends the word YOLO', () => {
    expect(
      someFn('supercalifragilisticexpialidocious')
    ).toEqual(
      'SUPERCALIFRAGILISTICEXPIALIDOCIOUSYOLO'
    );
  });
});
```
This unit test instead provides a good example of expected input/output and in combination with the test description, clearly illustrates (instead of duplicating) the source code.

Just to wrap up, I'm not saying we should never use variables or abstractions in tests. We absolutely do so in places and times where it makes sense (typically when it doesn't hinder explicit developer understanding). But I will strongly push for starting with static assertions as a baseline and abstractions as exception and not a rule in our unit tests.